backend/src/email/middlewares/email-template.middleware.ts

```diff
@@ -246,6 +248,52 @@ export const InitEmailTemplateMiddleware = (
     }
   }
 
+  const selectPhonebookListHandler = async (
+    req: Request,
+    res: Response
+  ): Promise<Response | void> => {
+    try {
+      const { campaignId } = req.params
+
+      const { list_id: listId } = req.body
+
+      // check if template exists
+      const template = await EmailTemplateService.getFilledTemplate(+campaignId)
+      if (template === null) {
+        throw new Error(
+          'Error: No message template found. Please create a message template before uploading a recipient file.'
+        )
+      }
+
+      const list = await PhonebookService.getPhonebookListById({
+        listId,
+      })
+      if (!list) throw new Error('Error: List not found')
+
+      const { s3Key, etag, filename } = list
+
+      // Store temp filename
+      await UploadService.storeS3TempFilename(+campaignId, filename)
+
+      // Enqueue upload job to be processed
+      await EmailTemplateService.enqueueUpload({
+        campaignId: +campaignId,
+        template,
+        s3Key,
+        etag,
+        filename,
+      })
+
+      return res.status(202).json({ list_id: listId })
+    } catch (e) {
+      // explicitly return a 500 to not block user flow but prompt them to upload an alternative csv
+      return res.status(500).json({
```

Comment:
If it's a handled error, I recommend we use something other than `500`

Reply:
it cant be a 4xx error as it's not within the user control, so something like `502` etc?

Reply:
after discussion, retaining this logic as we want to alert DD if this endpoint has issues, and bubble error up to UI to show to the user.

This is primarily due to the endpoint may be failing due to phonebook or postman upload CSV code. So we cannot suppress error as its not a solely phonebook error